source/data-modeling/associations.txt

```diff
@@ -515,3 +515,410 @@ method to remove all embedded ``Album`` documents from the ``Band`` class:
+association. This is useful when you want to use a different name for an
+association than the name of the class. The following example uses the
+``class_name`` macro to specify that an embedded association called ``records``
+represents the ``Album`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-name
+   :end-before: # end-custom-name
+
+Custom Keys
+~~~~~~~~~~~
+
+By default, {+odm+} uses the ``_id`` field of the parent class when looking up
+associations. You can specify a different field to use by specifying the
+``primary_key`` and ``foreign_key`` macros. The following example specifies a new
+primary and foreign key for the ``albums`` association on a ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-keys
+   :end-before: # end-custom-keys
+
+If you are specifying a ``has_and_belongs_to_many`` association, you can also
+set the ``inverse_primary_key`` and ``inverse_foreign_key`` macros. The
+``inverse_primary_key`` macro specifies the field on the local model that the
+remote model uses to look up the documents.
+The ``inverse_foreign_key`` macro specifies the field on the remote model
+that stores the values found in ``inverse_primary_key``.
+
+The following example specifies a new primary and foreign key for the
+``Band`` and ``Members`` classes in a ``has_and_belongs_to_many`` association:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-inverse-keys
+   :end-before: # end-custom-inverse-keys
+   :emphasize-lines: 9, 20
+
+Custom Scopes
+~~~~~~~~~~~~~
+
+You can set a specific scope on an association by using the ``scope`` parameter.
+The ``scope`` parameter determines which documents {+odm+} considers to be part
+of an association. A scoped association returns only documents that match the
+scope conditions when queried. You can set the ``scope`` to either a ``Proc`` with an arity
+of zero or a ``Symbol`` that references a named scope on the associated model.
+The following example sets custom scopes on associations in a ``Band`` class:
+
+.. literalinclude:: /includes/data-modeling/association-behaviors.rb
+   :language: ruby
+   :start-after: # start-custom-scope
+   :end-before: # end-custom-scope
+
+.. note::
+
+   You can add documents that do not match the scope conditions to an
+   association. {+odm+} saves the documents to the database and they will appear
+   in associated memory, however, the documents are not present when the
+   association is queried in the future.
+
+Validations
+~~~~~~~~~~~
+
+By default, {+odm+} validates that the children of any associations are present
+when it loads the association into memory by using the ``validates_associated``
+macro. This applies to the following association types:
```

Comment:
i: this sounds like it uses 'validates_associated' to load the association into memory

```suggestion
When {+odm+} loads an association into memory, by default, it uses the
``validates_associated``
macro to validate that any children are also present.
{+odm+} validates children for the following association types:
```